### PR TITLE
refactor (jkube-kit-generator-api): Migrate all tests from JUnit4 to JUnit5 (#1560)

### LINKED ISSUE #1560: (jkube-kit-generator-api): Replace JUnit4 with JUnit5
### Component

jkube-kit-generator-api

### Task description

Child of #1546

- Remove JUnit4 dependencies
- Add JUnit5 dependencies
- Refactor tests

### PR DESCRIPTION
**Pull request by anurag-rajawat**
## Description

Fixes #1560 

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [x] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/generator/api/src/test/java/org/eclipse/jkube/generator/api/support/AbstractPortsExtractorTest.java [hunk 1]
```diff
@@ -14,112 +14,120 @@
...
-    public void testNoProperty() {
-        Map<String, Integer> map = extractFromFile(null, getClass().getSimpleName() + ".yml");
-        assertNotNull(map);
-        assertEquals(0,map.size());
+    @DisplayName("add port to list")
+    @ParameterizedTest(name = "''{0}'' with port ''{1}''")
+    @MethodSource("ports")
+    void addPortToList(String portName, int expectedPort) {
+      Map<String, Integer> map = extractFromFile("vertx.config", getClass().getSimpleName() + "-pattern-values.yml");
+      assertThat(map).containsEntry(portName, expectedPort);
```
Comment (rohanKanojia) [inline jkube-kit/generator/api/src/test/java/org/eclipse/jkube/generator/api/support/AbstractPortsExtractorTest.java:116]: Why have you made these tests parameterized? I see only a constant input provided to test.

File: jkube-kit/generator/api/src/test/java/org/eclipse/jkube/generator/api/support/AbstractPortsExtractorTest.java [hunk 2]
```diff
@@ -14,112 +14,120 @@
...
-        };
-
-        for (int i = 0; i < testData.length; i +=2 ) {
-            assertEquals(testData[i+1], map.containsKey(testData[i]));
-        }
+    @ParameterizedTest(name = "''{0}'' should be ''{1}''")
+    @MethodSource("patterns")
+    void keyPatterns(String key, boolean expected) {
+      Map<String, Integer> map = extractFromFile("vertx.config", getClass().getSimpleName() + "-pattern-keys.yml");
+      assertThat(map.containsKey(key)).isEqualTo(expected);
```
Comment (rohanKanojia) [inline jkube-kit/generator/api/src/test/java/org/eclipse/jkube/generator/api/support/AbstractPortsExtractorTest.java:97]: I cannot understand why you're using a parameterized test to check map keys. 

Parameterized test are supposed to be used when there is a possibility of running same test with different input/outputs.

Comment (anurag-rajawat) [reply jkube-kit/generator/api/src/test/java/org/eclipse/jkube/generator/api/support/AbstractPortsExtractorTest.java]: I'm really sorry, it was due to some misunderstandings.
Fixed.

File: jkube-kit/generator/api/src/test/java/org/eclipse/jkube/generator/api/support/AbstractPortsExtractorTest.java (hunk 1 above)
Comment (anurag-rajawat) [reply jkube-kit/generator/api/src/test/java/org/eclipse/jkube/generator/api/support/AbstractPortsExtractorTest.java]: I'm really sorry, it was due to some misunderstandings related to testData.
Fixed.

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]
LGMT, thx!

**Pull request merged**